nautobot/project-static/js/forms.js

```diff
@@ -672,6 +696,39 @@ function initializeDynamicFilterForm(context){
         $("#dynamic-filter-form").submit()
     })
 
+    // On clear of filter form
+    this_context.find("#dynamic-filter-form, #default-filter form").on("reset", function(e){
+        console.log("hello")
+        e.preventDefault()
+        const urlParams = new URLSearchParams(window.location.search);
+        const saved_view_pk = urlParams.get('saved_view');
+        var saved_view_pk_query_string = ""
+        if (saved_view_pk != null){
+            saved_view_pk_query_string = "&saved_view=" + saved_view_pk
+        }
+        const sort_order = urlParams.getAll('sort');
+        var sort_order_query_string = ""
+        if (sort_order != null){
+            for (var i = 0; i < sort_order.length; i++){
+                appendage = "&sort=" + sort_order[i]
+                sort_order_query_string += appendage
+            }
+        }
+        const per_page = urlParams.get('per_page');
+        var per_page_query_string = ""
+        if (per_page_query_string != null) {
+            per_page_query_string = "per_page=" + per_page
+        }
+        const table_changes = urlParams.get('table_changes_pending');
+        var table_changes_query_string = ""
+        if (table_changes_query_string != null) {
+            table_changes_query_string = "&table_changes_pending=" + table_changes
+        }
```

Comment:
Might be cleaner/simpler to iterate over everything in `urlParams` and just delete from it anything that *isn't* saved_view/sort/per_page/table_changes_pending?

Reply:
Refactored to delete all except non_filter_params.
```
this_context.find("#dynamic-filter-form, #default-filter form").on("reset", function(e){
        e.preventDefault()
        // make two copies of url params
        const urlParams = new URLSearchParams(window.location.search);
        const newUrlParams = new URLSearchParams(window.location.search);
        // every query string that is non-filter-related
        const non_filter_params = ["saved_view", "sort", "per_page", "table_changes_pending"]
        for (const [key, value] of urlParams.entries()) {
            // remove filter params
            if (non_filter_params.includes(key) === false) {
                newUrlParams.delete(key, value)
            }
        }
        location.assign("?" + urlParams.toString())
    })
```